src/vic3/data/combat_units.rs

```diff
@@ -46,6 +48,101 @@ impl DbKind for OldCombatUnit {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct CombatUnit {}
+
+inventory::submit! {
+    ItemLoader::Normal(GameFlags::Vic3, Item::CombatUnit, CombatUnit::add)
+}
+
+impl CombatUnit {
+    pub fn add(db: &mut Db, key: Token, block: Block) {
+        db.add(Item::CombatUnit, key, block, Box::new(Self {}));
+    }
+}
+
+impl DbKind for CombatUnit {
+    // This whole item type is undocumented.
+    fn validate(&self, key: &Token, block: &Block, data: &Everything) {
+        data.verify_exists(Item::Localization, key);
+        let loca = format!("{key}_desc");
+        data.verify_exists_implied(Item::Localization, &loca, key);
+
+        let mut vd = Validator::new(block, data);
+        vd.req_field("group");
+        vd.req_field("combat_unit_image");
+
+        vd.field_item("group", Item::CombatUnitGroup);
+        vd.field_validated_block("battle_modifier", |block, data| {
+            let vd = Validator::new(block, data);
+            validate_modifs(block, data, ModifKinds::Unit, vd);
+        });
+        vd.field_validated_block("upkeep_modifier", |block, data| {
+            let vd = Validator::new(block, data);
+            // The upkeep modifier gets applied to the country so it can
+            // actually take a variety of `ModifKinds`.
+            validate_modifs(block, data, ModifKinds::all(), vd);
+        });
+        vd.field_validated_block("formation_modifier", |block, data| {
+            let vd = Validator::new(block, data);
+            validate_modifs(block, data, ModifKinds::MilitaryFormation, vd);
+        });
+        vd.field_validated_key_block("can_build_conscript", |key, block, data| {
+            let mut sc = ScopeContext::new(Scopes::Country, key);
+            validate_trigger(block, data, &mut sc, Tooltipped::No);
+        });
+        vd.field_list_items("unlocking_technologies", Item::Technology);
+        let mut seen_unconditional = false;
+        vd.multi_field_validated_key_block("combat_unit_image", |key, block, data| {
+            if seen_unconditional {
+                let msg = "there was a previous `combat_unit_image` without a trigger, so this one will not be used";
```

Comment:
would be better if can provide the location of that previous block saved in `seen_unconditional: Option<Token/Block>` and then point to it with `loc(seen_block, "from here")`. Also consider suggesting adding something like "; move that to the end as the fallback".